### PR TITLE
Added locking mechanism

### PR DESCRIPTION
**Pull request by wachterjohannes**
Integrates locking mechanism from https://github.com/php-task/php-task/pull/35

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -43,14 +56,46 @@ public function getConfigTreeBuilder()
...
+                                    ->addDefaultsIfNotSet()
+                                    ->children()
+                                        ->scalarNode('directory')->defaultValue('%kernel.cache_dir%/tasks')->end()
+                                    ->end()
+                                ->end()
+                            ->end()
                         ->end()
                     ->end()
                 ->end()
+                ->enumNode('mode')->values(['off', 'listener'])->defaultValue('off')->end()
```
Comment (alexander-schranz) [inline src/DependencyInjection/Configuration.php]: duplicated

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

File: composer.json [hunk 2]
```diff
@@ -11,9 +11,10 @@
     ],
     "require": {
         "php": "~5.5 || ~7.0",
-        "php-task/php-task": "dev-develop",
+        "php-task/php-task": "dev-feature/lock",
```
Comment (alexander-schranz) [inline composer.json]: @wachterjohannes fix

**Comment by wachterjohannes** [comment]
@alexander-schranz done

**Pull request merged**